### ISSUE TITLE
Split test definitions across multiple files

### ISSUE THREAD

**Issue by fnogatz** [fnogatz/tap#3]
Let's say we want to split our test definitions across multiple files. For example, we could want to enhance the example given in the [documentation](https://github.com/mndrix/tap/blob/42d822de3f0fb2f3aaeafeb200b683ce01fc27ac/README.md#macros-that-write-tests) by defining tests to get the length of lists of atoms in another file than the tests to get the length of lists of integers. So at the end we have a file `int_list.pl` and `atom_list.pl`:

*atom_list.pl*:
```prolog
[a,b,c] -> 3.
[a,b] -> 2.
```

*int_list.pl*:
```prolog
[1,2,3] -> 3.
[1,2] -> 2.
```

I tried many variations to load these tests dynamically (i.e., adding a new file `string_list.pl` automatically results in more tests run) but have not found a solution yet. A simple approach would be:

*test.pl*:
```prolog
term_expansion(List -> Length, (Head :- Test)) :-
    format(atom(Head), 'length(~w, ~w)', [List, Length]),
    Test = (
        length(List, Len),
        Len = Length
    ),
    tap:register_test(Head).

term_expansion(run_length_tests, []) :-
    directory_files('.', Files),
    forall(
        (member(File, Files), \+member(File, ['.','..','test.pl'])),
        consult(File)
    ).

:- use_module(library(tap)).
run_length_tests.
```

Unfortunately this only runs though one file, `int_list.pl`:

```bash
> swipl -q -t main -f test.pl
TAP version 13
1..2
ok 1 - length([1,2,3], 3)
ok 2 - length([1,2], 2)
```

Any ideas how to extend this example to dynamically load tests specified in separate files within a directory?

**Comment by mndrix** [comment]
At first glance, your code seems like it should work.  If you run the following, do you see all the `'length(...,...) :- ...` clauses that you expect?

```
> swipl -q -t listing -f test.pl
```

That would help us to know if the problem is in the macro expansion or if tap is just not running the tests correctly.

**Comment by fnogatz** [comment]
Uh, just learned a new simple way for debugging by using `-t listing`, thank you :)

Apparently [these lines](https://github.com/mndrix/tap/blob/master/prolog/tap.pl#L34-L41) in `tap.pl` create multiple `main/0` predicates:

```prolog
main :-
	tap_header(2),
	tap_call('length([1,2,3], 3)', 1, A),
	tap_call('length([1,2], 2)', A, _).
main :-
	tap_header(2),
	tap_call('length([a,b,c], 3)', 1, A),
	tap_call('length([a,b], 2)', A, _).
main :-
	tap_header(0).
```

**Comment by mndrix** [comment]
That makes sense now that I see it.  We probably need a stricter definition for [term_wants_tap_expansion](https://github.com/mndrix/tap/blob/42d822de3f0fb2f3aaeafeb200b683ce01fc27ac/prolog/tap.pl#L27-L30). It should probably detect whether the file being loaded has directly imported `library(tap)`.  The separate test files would fail that test but the main file would pass.

[Similar code in library(func)](https://github.com/mndrix/func/blob/7c4cd36122a21f64e26c67d31d07653b4d124d47/prolog/func.pl#L12-L17) might be a starting point for what we want here.

**Comment by mndrix** [comment]
Included in [v0.4.0](https://github.com/mndrix/tap/releases/tag/v0.4.0).
